lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -62,20 +78,28 @@ def get(self):
         data = self.client.read(self.secret)
         if data is None:
             raise AnsibleError("The secret %s doesn't seem to exist" % self.secret)
-        else:
-            return data['data']['value']
+        
+        if self.secret_field=='': # secret was specified with trailing ':'
+            return data['data']
+        
+        if self.secret_field not in data['data']:
+            raise AnsibleError("The secret %s does not contain the field '%s'. " % (self.secret, self.secret_field))
+        
+        return data['data'][self.secret_field]
 
 
 class LookupModule(LookupBase):
 
     def run(self, terms, variables, **kwargs):
-
         vault_args = terms[0].split(' ')
         vault_dict = {}
         ret = []
 
         for param in vault_args:
-            key, value = param.split('=')
+            try:
+                key, value = param.split('=')
+            except ValueError as e:
+                raise AnsibleError("hashi_vault plugin needs key=value pairs, but received %s" % terms)
```

Comment:
This may be true; I have no time to test that right now. However, it is not a regression as the existing implementation does not do anything special either. So it's not related to this PR, IMHO. 

I do not currently have time to work on this, but the improvements currently implemented are self-contained. 

Can we address the list-scenario in a separate issue?